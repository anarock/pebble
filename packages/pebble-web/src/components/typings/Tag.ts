import React from "react";
import { SetDifference } from "utility-types";
import { Colors } from "pebble-shared/dist/theme/typings/colors";
import { Interpolation } from "./emotionCustom";

export interface TagProps {
  label: React.ReactNode;
  color: SetDifference<keyof Colors, "white">;
  styles?: Interpolation;
  onClose?: () => void;
}
