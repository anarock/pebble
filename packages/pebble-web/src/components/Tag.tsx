import * as React from "react";
import { colors } from "pebble-shared";
import { tagStyle, iconClass } from "./styles/Tag.styles";
import { TagProps } from "./typings/Tag";

import Ink from "react-ink";

const Tag: React.FunctionComponent<TagProps> = ({
  label,
  color,
  onClose,
  styles
}) => {
  return (
    <div
      className={(!!onClose && "__pebble__tag__with__close") || ""}
      css={[tagStyle, styles]}
      style={{
        color: colors[color].base,
        // @ts-ignore
        backgroundColor: colors[color].lightest || colors[color].light
      }}
    >
      {label}{" "}
      {onClose && (
        <i onClick={onClose} className="pi pi-close" css={iconClass}>
          <Ink />
        </i>
      )}
    </div>
  );
};

export default Tag;
