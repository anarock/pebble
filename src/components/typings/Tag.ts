import { SetDifference } from "utility-types";
import { Colors } from "../../theme/typings/colors";

export interface TagProps {
  label: React.ReactNode;
  color: SetDifference<keyof Colors, "white">;
  className?: string;
  onClose?: () => void;
}
