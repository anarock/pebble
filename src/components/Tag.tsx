import * as React from "react";
import { colors } from "../theme";
import { tagStyle } from "./styles/Tag.styles";
import { TagProps } from "./typings/Tag";
import { cx } from "emotion";
import Ink from "react-ink";

const Tag: React.SFC<TagProps> = ({ label, color, onClose }) => {
  const wrapperClassName = cx(tagStyle, {
    __pebble__tag__with__close: !!onClose
  });
  return (
    <div
      className={wrapperClassName}
      style={{
        color: colors[color].base,
        backgroundColor: colors[color].lightest || colors[color].light
      }}
    >
      {label}{" "}
      {onClose && (
        <i onClick={onClose} className="pi pi-close">
          <Ink />
        </i>
      )}
    </div>
  );
};

export default Tag;
