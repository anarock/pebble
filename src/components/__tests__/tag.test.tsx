import * as React from "react";
import { mount, shallow } from "enzyme";
import sinon from "sinon";
import Tag from "../Tag";

describe("Component: Tag", () => {
  test("snapshot", () => {
    const tag = shallow(<Tag label="Anarock" color="violet" />);
    expect(tag).toMatchSnapshot();

    const tag2 = shallow(
      <Tag label="Anarock" color="emerald" onClose={() => {}} />
    );
    expect(tag2).toMatchSnapshot();
  });

  test("onClose is triggered correctly", () => {
    const spy = sinon.spy();

    const tag = mount(<Tag label="Anarock" color="emerald" onClose={spy} />);
    tag.find("i").simulate("click");
    expect(spy.calledOnce).toBeTruthy();
  });
});
