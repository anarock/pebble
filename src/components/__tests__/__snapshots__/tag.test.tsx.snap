// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tag snapshot 1`] = `
.emotion-0 {
  font-weight: 500;
  color: #101721;
  font-size: 10px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: initial;
  -ms-flex-line-pack: initial;
  align-content: initial;
  border-radius: 3px;
  padding: 0 15px;
  height: 25px;
  text-transform: uppercase;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 10px;
}

.emotion-0.__pebble__tag__with__close {
  padding-right: 0;
}

.emotion-0 i {
  font-size: 7px;
  margin-left: 5px;
  height: inherit;
  padding: 9px 10px;
  cursor: pointer;
  position: relative;
}

<div
  className="emotion-0"
  style={
    Object {
      "backgroundColor": "#E8E8FF",
      "color": "#6161FF",
    }
  }
>
  Anarock
   
</div>
`;

exports[`Component: Tag snapshot 2`] = `
.emotion-0 {
  font-weight: 500;
  color: #101721;
  font-size: 10px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: initial;
  -ms-flex-line-pack: initial;
  align-content: initial;
  border-radius: 3px;
  padding: 0 15px;
  height: 25px;
  text-transform: uppercase;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 10px;
}

.emotion-0.__pebble__tag__with__close {
  padding-right: 0;
}

.emotion-0 i {
  font-size: 7px;
  margin-left: 5px;
  height: inherit;
  padding: 9px 10px;
  cursor: pointer;
  position: relative;
}

<div
  className="emotion-0 __pebble__tag__with__close"
  style={
    Object {
      "backgroundColor": "#DBF3E7",
      "color": "#10AC60",
    }
  }
>
  Anarock
   
  <i
    className="icon-close"
    onClick={[Function]}
  >
    <e
      background={true}
      className="ink"
      duration={1000}
      opacity={0.25}
      radius={150}
      recenter={true}
    />
  </i>
</div>
`;
